packages/material-ui/src/TextareaAutosize/TextareaAutosize.js

```diff
@@ -68,17 +68,18 @@ const TextareaAutosize = React.forwardRef(function TextareaAutosize(props, ref)
 
     // Take the box sizing into account for applying this value as a style.
     const outerHeightStyle = outerHeight + (boxSizing === 'border-box' ? padding + border : 0);
+    const overflow = Math.abs(outerHeight - innerHeight) <= 1;
 
     setState(prevState => {
       // Need a large enough different to update the height.
       // This prevents infinite rendering loop.
       if (
-        outerHeightStyle > 0 &&
-        Math.abs((prevState.outerHeightStyle || 0) - outerHeightStyle) > 1
+        (outerHeightStyle > 0 &&
+          Math.abs((prevState.outerHeightStyle || 0) - outerHeightStyle) > 1) ||
+        prevState.overflow !== overflow
       ) {
         return {
-          innerHeight,
-          outerHeight,
+          overflow,
```

Comment:
Sorry for so many comments.

We're returing `overflow` even if the below condition is true.
```
 (outerHeightStyle > 0 &&
        Math.abs((prevState.outerHeightStyle || 0) - outerHeightStyle) > 10)
```

Don't you think `overflow` state should only be updated when `prevState.overflow !== overflow` is true ?

Reply:
We are using a single state. I wouldn't expect any difference.

Reply:
Umm Okay!